FreeRTOS-Plus/Test/CBMC/proofs/UDP/vProcessGeneratedUDPPacket/vProcessGeneratedUDPPacket_harness.c

```diff
@@ -0,0 +1,92 @@
+/* Standard includes. */
+#include <stdint.h>
+#include <stdio.h>
+
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+#include "list.h"
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+
+#include "FreeRTOS_ARP.h"
+#include "FreeRTOS_UDP_IP.h"
+//#include "FreeRTOS_TCP_IP.h"
+#include "FreeRTOS_IP_Private.h"
+
+
+
+uint16_t usGenerateChecksum( uint16_t usSum, const uint8_t * pucNextData, size_t uxByteCount ) 
+{
+	uint16_t usChecksum;
+
+	/* Return any random value of checksum since it does not matter for CBMC checks. */
+	return usChecksum;
+}
+
+uint16_t usGenerateProtocolChecksum( const uint8_t * const pucEthernetBuffer, size_t uxBufferLength, BaseType_t xOutgoingPacket )
+{
+	uint16_t usProtocolChecksum;
+
+	/* Return random value of checksum since it does not matter for CBMC checks. */
+	return usProtocolChecksum;
+}
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+void vARPRefreshCacheEntry( const MACAddress_t * pxMACAddress, const uint32_t ulIPAddress ) { }
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+void vARPGenerateRequestPacket( NetworkBufferDescriptor_t * const pxNetworkBuffer )         { }
+
+/* This function has been tested separately. Therefore, we assume that the implementation is correct. */
+eARPLookupResult_t eARPGetCacheEntry( uint32_t *pulIPAddress, MACAddress_t * const pxMACAddress )
+{
+	eARPLookupResult_t eResult;
+	return eResult;
+}
+
+/* Implementation of safe malloc. Return a NULL or the expected buffer indeterministically. */
+void *safeMalloc(size_t xWantedSize ){
+	if(xWantedSize == 0){
+		return NULL;
+	}
+	uint8_t byte = nondet_uint8_t();
+	return byte ? malloc(xWantedSize) : NULL;
+}
+
+
+/* Abstraction of pxGetNetworkBufferWithDescriptor. We assume it to be correctly implemented. */
+NetworkBufferDescriptor_t *pxGetNetworkBufferWithDescriptor( size_t xRequestedSizeBytes, TickType_t xBlockTimeTicks ){
```

Comment:
These abstractions (stubs) are is several of your pull requests today.  I would put them into CBMC/stubs or CBMC/sources, one stub per file, so the stubs can be consistently reused in all of your proofs.

Reply:
Done so. Thanks.